server/resp/src/main/java/org/infinispan/server/resp/commands/cluster/NODES.java

```diff
@@ -0,0 +1,196 @@
+      return readNodeInformation(hash.getMembers(), executor)
+            .thenApply(information -> {
+               // The response is a bulk string, each line contains information about one node.
+               // The format of each line is:
+               // <id> <ip:port@cport[,hostname[,auxiliary_field=value]*]> <flags> <master> <ping-sent> <pong-recv> <config-epoch> <link-state> <slot> <slot> ... <slot>
+               // More information about each field is available at the command link.
+               // Since some of that information doesn't make sense for Infinispan, so we add some simple value since it is required.
+               StringBuilder response = new StringBuilder();
+               boolean first = true;
+
+               Address local = ecm.getAddress();
+               for (Address member : hash.getMembers()) {
+                  if (!first) {
+                     response.append('\n');
+                  }
+                  first = false;
+                  boolean isMyself = member.equals(local);
+                  IntSet owner = IntSets.from(hash.getSegmentsForOwner(member));
+                  String initial = information.get(member);
+                  String health = "connected";
+
+                  if (initial != null) {
+                     // We were able to connect the node.
+                     // This contains the <id> <ip:port@cport[,hostname[,auxiliary_field=value]*]>
+                     response.append(initial);
+                     if (isMyself) {
+                        response.append("myself,");
+                     }
+                  } else {
+                     // We could not retrieve information from the node.
+                     // We add the information as if was disconnected.
+                     response.append(member).append(' ');
+                     response.append(member).append(':').append(0).append('@').append(0);
+                     response.append(",,shard-id=").append(member).append(' ');
+
+                     if (isMyself) {
+                        response.append("myself,");
+                     } else {
+                        response.append("fail?,");
+                     }
+                     health = "disconnected";
+                  }
+
+                  response.append("master").append(' ');
+                  response.append('-').append(' ');
+                  response.append('0').append(' ');
+                  response.append(Instant.now().getEpochSecond()).append(' ');
+                  response.append(topology.getTopologyId()).append(' ');
+                  response.append(health).append(' ');
+                  serializeSegments(response, owner);
+               }
+               return "$" + response.length() + CRLF + response + CRLF;
+            });
+   }
+
+   private static void serializeSegments(StringBuilder response, IntSet ranges) {
+      BitSet bitSet = BitSet.valueOf(ranges.toBitSet());
```

Comment:
This should be able to just use the ranges.iterator instead. But this change is not required as it is a pretty small optimization I suspect.

Reply:
Another option is adding nextSetBit method to IntSet which sounds like a good idea to me. It should be pretty trivial to add the method to the various implementations.

Reply:
Added the `nextSetBit` method, and updated here and the SHARDS command.